app/controllers/enrolment/ReservationController.java

```diff
@@ -164,34 +163,31 @@ public CompletionStage<Result> removeReservation(long id, Http.Request request)
     }
 
     @Restrict({ @Group("ADMIN") })
-    public Result findAvailableMachines(Long reservationId) throws ExecutionException, InterruptedException {
+    public Result findAvailableMachines(Long reservationId, Long roomId)
+        throws ExecutionException, InterruptedException {
         var reservation = DB.find(Reservation.class, reservationId);
         if (reservation == null) {
             return notFound();
         }
+        if (DB.find(ExamRoom.class, roomId) == null) {
```

Comment:
Oisko näistä kahesta If:stä voinu tehä samanlaiset? Kun nyt on kaks eri toteutusta samasta asiasta peräkkäin?